docs/proposals/20220824-persistentpodstate-custom-workload-support.md

```diff
@@ -0,0 +1,200 @@
+---
+title: PersistentPodState custom workload support
+authors:
+- "@baxiaoshi"
+reviewers:
+- ""
+creation-date: 2022-08-24 
+last-updated: 2022-08-24
+status: 
+---
+
+# PersistentPodState custom workload support
+Table of Contents
+=================
+
+* [Motivation](#motivation)
+* [Proposal](#proposal)
+    * [1.add dynamic watch](#1add-dynamic-watch)
+    * [2.add workload finder](#2add-workload-finder)
+## Motivation
+PPS(PersistentPodState) currently has support for standard statefulset and kruise's statefulSet. However, some companies will customize workloads(statefulset like) with stable identification pods,  support such workloads, and add support for custom workloads to pps.
+## Proposal
+Modify PPS controller logic to support custom statefulset like workload：
+1.add dynamic watch: for watch custom statefulset like workload, automatically create pps if pps persistent annotations are found
+2.add workload finder: used to find workload according to the target of PPS and build ScaleAndSelector object
+### 1.add dynamic watch
+client-go supports dynamic watch, but requires gvr or gvk to provide the resources. There are two ways to get the resources in the cluster that need to be watched：
+
+1）Get the APIResourceList  in the cluster through the ServerPreferredResources interface of the dynamic client. Because we can't distinguish those workloads, we can only watch all of them here, which will have a lot of unnecessary watch overhead.
+2）Find the corresponding workload by the reverse of the pod's ownerReferences. There is a time delay.
+
+It is recommended to use method 2 to get the owner's gvk by reverse lookup and create Informer if there is no watch before.
```

Comment:
PPS rely on the ordinal maintenance of statefulset, and actually pps can works for statefulset-like workload how to ensure custom workload can provide similar capability and avoid incorrect usage? maybe adding a new config in kruise-configuration configmaps

Reply:
Add PPS_Watch_Custom_Workload_WhiteList in kruise-configuration to declare the Workload that needs to be watched